tests/sqlite3/test-sqlite3.cpp

```diff
@@ -641,6 +641,21 @@ TEST_CASE("SQLite DDL with metadata", "[sqlite][ddl]")
     CHECK(ddl_t3_found == false);
 }
 
+TEST_CASE("SQLite DDL symmetry", "[sqlite][ddl][symmetry]")
+{
+    soci::session sql(backEnd, connectString);
+    test_limit(sql, DdlTester<double>       (soci::db_double, std::numeric_limits<double>::max(),        false));
+    test_limit(sql, DdlTester<double>       (soci::db_double, std::numeric_limits<double>::quiet_NaN(),  true));
```

Comment:
So this is the _only_ case when we pass `true` as `expectNull`? I'd just remove it and simplify things by removing all the related checks, it seems like a huge overkill to add all this code just to check that NaN is getting returned as null (and I'm not even sure if it's the desired behaviour!).

Reply:
This is a SQLite specific behavior